library_generation/owlbot/synthtool/languages/java.py

```diff
@@ -0,0 +1,691 @@
+    diregapic: bool = False,
+    preserve_gapic: bool = False,
+):
+    """Helper function to execution the common generation cleanup actions.
+
+    Fixes headers for protobuf classes and generated gRPC stub services. Copies
+    code and samples to their final destinations by convention. Runs the code
+    formatter on the generated code.
+
+    Args:
+        service (str): Name of the service.
+        version (str): Service API version.
+        library (Path): Path to the temp directory with the generated library.
+        package_pattern (str): Package name template for fixing file headers.
+        suffix (str, optional): Suffix that the generated library folder. The
+            artman output differs from bazel's output directory. Defaults to "".
+        destination_name (str, optional): Override the service name for the
+            destination of the output code. Defaults to the service name.
+        preserve_gapic (bool, optional): Whether to preserve the gapic directory
+            prefix. Default False.
+    """
+
+    if destination_name is None:
+        destination_name = service
+
+    cloud_prefix = "cloud-" if cloud_api else ""
+    package_name = package_pattern.format(service=service, version=version)
+    fix_proto_headers(
+        library / f"proto-google-{cloud_prefix}{service}-{version}{suffix}"
+    )
+    fix_grpc_headers(
+        library / f"grpc-google-{cloud_prefix}{service}-{version}{suffix}", package_name
+    )
+
+    if preserve_gapic:
+        s.copy(
+            [library / f"gapic-google-{cloud_prefix}{service}-{version}{suffix}/src"],
+            f"gapic-google-{cloud_prefix}{destination_name}-{version}/src",
+            required=True,
+        )
+    else:
+        s.copy(
+            [library / f"gapic-google-{cloud_prefix}{service}-{version}{suffix}/src"],
+            f"google-{cloud_prefix}{destination_name}/src",
+            required=True,
+        )
+
+    s.copy(
+        [library / f"grpc-google-{cloud_prefix}{service}-{version}{suffix}/src"],
+        f"grpc-google-{cloud_prefix}{destination_name}-{version}/src",
+        # For REST-only clients, like java-compute, gRPC artifact does not exist
+        required=(not diregapic),
+    )
+    s.copy(
+        [library / f"proto-google-{cloud_prefix}{service}-{version}{suffix}/src"],
+        f"proto-google-{cloud_prefix}{destination_name}-{version}/src",
+        required=True,
+    )
+
+    if preserve_gapic:
+        format_code(f"gapic-google-{cloud_prefix}{destination_name}-{version}/src")
+    else:
+        format_code(f"google-{cloud_prefix}{destination_name}/src")
+    format_code(f"grpc-google-{cloud_prefix}{destination_name}-{version}/src")
+    format_code(f"proto-google-{cloud_prefix}{destination_name}-{version}/src")
+
+
+def _merge_release_please(destination_text: str):
+    config = yaml.safe_load(destination_text)
+    if "handleGHRelease" in config:
+        return destination_text
+
+    config["handleGHRelease"] = True
```

Comment:
nit: change string `handleGHRelease` to a variable.

Reply:
Done